### PR TITLE
Rover: add guided velocity controler

### PR DESCRIPTION
**Pull request by khancyr**
This PR add the support of velocity control in guided mode (msg SET_POSITION). It is based on @rmackay9 previous works.
It has been tested on SITL with ROS and in gazebo

### CODE REVIEW

**Comment by OXINARF** [comment]
Just mentioning that this PR includes the commits from #5734 so be careful when merging this so that we don't end up with duplicate commits in master.

**Comment by gmorph** [comment]
Hi khancyr.  Can you please remove the commits from #5734 as I have already merged that in.

Thanks, Grant.

**Comment by khancyr** [comment]
@gmorph done!

**Comment by gmorph** [comment]
We need to test this A LOT.  @khancyr - have you done much SITL and real world testing with this?
Thanks, Grant.

**Comment by khancyr** [comment]
@gmorph I used this with SITL and Gazebo mostly on skid steer configuration: see https://www.youtube.com/watch?v=o0zC7NgNBgE&index=8&list=PL6sCNLbHuYxYyFWLTq3E-R2cV1CrEoYOe for example. I don't have access to rover anymore so I can't test on pixhawk ...

**Comment by khancyr** [comment]
rebased on master

**Comment by gmorph** [comment]
I need to test this - sorry for the delay.

**Comment by khancyr** [comment]
rebased !

**Comment by khancyr** [comment]
rebased

File: APMrover2/APMrover2.cpp [hunk 1]
```diff
@@ -565,11 +569,12 @@ void Rover::update_navigation()
...
     case RTL:
         // no loitering around the wp with the rover, goes direct to the wp position
-        calc_lateral_acceleration();
-        calc_nav_steer();
         if (verify_RTL()) {
             SRV_Channels::set_output_scaled(SRV_Channel::k_throttle, g.throttle_min.get());
             set_mode(HOLD);
+        } else {
+            calc_lateral_acceleration();
+            calc_nav_steer();
```
Comment (gmorph) [inline APMrover2/APMrover2.cpp]: These changes and the one below have nothing to do guided velocity controller.  They are just a re-order.  Good idea but I would of preferred this to be in a separate commit. In fact if this was in a separate PR a quick test and I could of merged this straight in.  But as it stands it adds time and complexity to the review process.

File: APMrover2/GCS_Mavlink.cpp [hunk 2]
```diff
@@ -1268,36 +1309,68 @@ void GCS_MAVLINK_Rover::handleMessage(mavlink_message_t* msg)
             }
 
             bool pos_ignore = packet.type_mask & MAVLINK_SET_POS_TYPE_MASK_POS_IGNORE;
+            bool vel_ignore = packet.type_mask & MAVLINK_SET_POS_TYPE_MASK_VEL_IGNORE;
+            bool acc_ignore = packet.type_mask & MAVLINK_SET_POS_TYPE_MASK_ACC_IGNORE;
+
+            // record the time where the last message arrived
```
Comment (gmorph) [inline APMrover2/GCS_Mavlink.cpp]: "when" the last message arrived - not "where"

File: APMrover2/GCS_Mavlink.cpp [hunk 3]
```diff
@@ -1268,36 +1309,68 @@ void GCS_MAVLINK_Rover::handleMessage(mavlink_message_t* msg)
...
             if (!pos_ignore) {
-                Location loc = rover.current_loc;
                 switch (packet.coordinate_frame) {
                 case MAV_FRAME_BODY_NED:
                 case MAV_FRAME_BODY_OFFSET_NED: {
                     // rotate from body-frame to NE frame
-                    float ne_x = packet.x*rover.ahrs.cos_yaw() - packet.y*rover.ahrs.sin_yaw();
-                    float ne_y = packet.x*rover.ahrs.sin_yaw() + packet.y*rover.ahrs.cos_yaw();
+                    float ne_x = packet.x * rover.ahrs.cos_yaw() - packet.y * rover.ahrs.sin_yaw();
+                    float ne_y = packet.x * rover.ahrs.sin_yaw() + packet.y * rover.ahrs.cos_yaw();
```
Comment (gmorph) [inline APMrover2/GCS_Mavlink.cpp]: AGAIN whitespace changes should be in a separate commit.

File: APMrover2/GCS_Mavlink.cpp (hunk 2 above)
Comment (khancyr) [inline APMrover2/GCS_Mavlink.cpp]: done!

File: APMrover2/APMrover2.cpp (hunk 1 above)
Comment (khancyr) [inline APMrover2/APMrover2.cpp]: I should have done that ! I extract them now

File: APMrover2/GCS_Mavlink.cpp (hunk 3 above)
Comment (khancyr) [inline APMrover2/GCS_Mavlink.cpp]: it was in a function I modified ... so a PR for 2 spaces ... I know somebody that will to kill me ;-P

**Comment by khancyr** [comment]
rebased, correct the last comment : extract unrelated changes, extract style only changes
